Updated installation instructions

capmisson opened the pull request:
I updated the README with installation instructions from scratch on a Debian 9, done with mysql.
I did prepare a script for it but might me more interesting like this to have clear what is needed. 
Some improvements can be done to avoid change user back and forward.

Added a example.com.conf as a example configuration file for Pixelfed following what I got from some of your examples and examples provided by users.

Finally fully working at https://anartist.world

capmisson committed:
First changes

capmisson committed:
More changes

capmisson committed:
example.com.conf

dansup commented on README.md:
````diff
@@ -38,28 +40,351 @@ testing and development.
+```
+
+Install Erlang Solutions repo
+```sh
````
Why is elixir in this guide? It is not required for pixelfed

dansup commented on README.md:
````diff
@@ -38,28 +40,351 @@ testing and development.
+$ sudo apt-get update && ${SUDO} apt-get install yarn
+```
+
+Install htop, bmon, mc and misc tools required by pixelfed
````
`htop`, `bmon` and `mc` are not required by pixelfed

capmisson replied:
Updated

dansup commented on README.md:
````diff
@@ -38,28 +40,351 @@ testing and development.
+
+Install PHP 7.2 FPM + additional extensions requiredd by pixelfed
+```sh
+$ sudo apt-get install -y php7.2-fpm php7.2 php7.2-common php7.2-cli php7.2-gd php7.2-mbstring php7.2-xml php7.2-json php7.2-bcmath php7.2-pgsql php7.2-curl
````
`php7.2-pgsql` is not required by pixelfed

dansup commented on README.md:
````diff
@@ -38,28 +40,351 @@ testing and development.
+
+Install svgo required by pixelfed, also curl
+```sh
+$ sudo npm install -g svgo
````
`svgo` is a deprecated dependency, it can be removed.

capmisson replied:
Changed

dansup commented on README.md:
````diff
@@ -38,28 +40,351 @@ testing and development.
+$ sudo apt-get install -y php7.2-fpm php7.2 php7.2-common php7.2-cli php7.2-gd php7.2-mbstring php7.2-xml php7.2-json php7.2-bcmath php7.2-pgsql php7.2-curl
+```
+
+Install Nginx, Redis and Postgres
````
`postgres` is not officially supported yet, please use remove it and use `mysql`

capmisson replied:
Changed

capmisson committed:
Edited changes requested

Took out elixir, htop, bmon, mc, php7.2-pgsql, svgo and postgresql from dependences

capmisson commented:
Did all requested changes, I think it can be updated giving the username who installs access to www-data group so wouldn't be needed to change property for files. But still this is a good starting point.

hellcp commented:
@dansup maybe considering that guide got both very distro-specific, and makes the readme very long, it would be better to include links to github wiki pages explaining the configuration for multiple platforms.

capmisson commented:
@hellcp yep that's true, maybe it makes more sense to put it on https://github.com/pixelfed/docs and start from there? Didn't mean to make it that long but if it's included instructions from scratch it does...

hellcp commented:
@capmisson it might not be a bad idea

capmisson commented:
@dansup if you prefer me to pull a request on https://github.com/pixelfed/docs instead of the official README so we can redirect specific Debian 9 users to there, just let me know.

Thanks

dansup commented:
@capmisson Yes please move this to https://github.com/pixelfed/docs/blob/master/master/installation.md, thanks!
